app/main.py

```diff
@@ -1,3 +1,64 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(
+                km=self.km + other.km
+            )
+        return Distance(
+            km=self.km + other
+        )
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km = self.km + other.km
+        elif isinstance(other, (int, float)):
+            self.km = self.km + other
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        return Distance(
+            km=self.km * other
+        )
+
+    def __truediv__(self, other: int | float) -> Distance:
```

Comment:
maybe add check for zero value
if other == 0:
    raise ValueError("Cannot divide by zero.")
    
    but if not - ok. will approve, all other's great